solidity/test/LPRewardsTest.js

```diff
@@ -0,0 +1,219 @@
+      const initialWrappedTokenStakerBallance1 = web3.utils.toBN(10000).mul(tokenDecimalMultiplier)
+      const initialWrappedTokenStakerBallance2 = web3.utils.toBN(20000).mul(tokenDecimalMultiplier)
+
+      await mintAndApproveWrappedTokens(
+        wrappedToken,
+        lpRewards.address,
+        staker1,
+        initialWrappedTokenStakerBallance1
+      )
+      await mintAndApproveWrappedTokens(
+        wrappedToken,
+        lpRewards.address,
+        staker2,
+        initialWrappedTokenStakerBallance2
+      )
+
+      let wrappedTokenBalance = await wrappedToken.balanceOf(lpRewards.address)
+      expect(wrappedTokenBalance).to.eq.BN(0)
+
+      let wrappedTokenStakerBalance1 = await wrappedToken.balanceOf(staker1)
+      expect(wrappedTokenStakerBalance1).to.eq.BN(initialWrappedTokenStakerBallance1)
+      let wrappedTokenStakerBalance2 = await wrappedToken.balanceOf(staker2)
+      expect(wrappedTokenStakerBalance2).to.eq.BN(initialWrappedTokenStakerBallance2)
+
+      await lpRewards.stake(web3.utils.toBN(4000).mul(tokenDecimalMultiplier), {
+        from: staker1,
+      })
+      await lpRewards.stake(web3.utils.toBN(5000).mul(tokenDecimalMultiplier), {
+        from: staker2,
+      })
+
+      wrappedTokenBalance = await wrappedToken.balanceOf(lpRewards.address)
+      // 4,000 + 5,000 = 9,000
+      expect(wrappedTokenBalance).to.eq.BN(
+        web3.utils.toBN(9000).mul(tokenDecimalMultiplier)
+      )
+
+      wrappedTokenStakerBalance1 = await wrappedToken.balanceOf(staker1)
+      // 10,000 - 4,000 = 6,000
+      expect(wrappedTokenStakerBalance1).to.eq.BN(
+        web3.utils.toBN(6000).mul(tokenDecimalMultiplier)
+      )
+
+      wrappedTokenStakerBalance2 = await wrappedToken.balanceOf(staker2)
+      // 20,000 - 5,000 = 15,000
+      expect(wrappedTokenStakerBalance2).to.eq.BN(
+        web3.utils.toBN(15000).mul(tokenDecimalMultiplier)
+      )
+    })
+  })
+
+  describe("tokens distribution", () => {
+    const precision = 1
+
+    it("should be possible to check earned rewards", async () => {
+      const keepRewards = new BN(10000).mul(tokenDecimalMultiplier)
+      const wrappedTokenStakerBallance = web3.utils
+        .toBN(10000)
+        .mul(tokenDecimalMultiplier)
+
+      await fundKEEPReward(lpRewards.address, keepRewards)
+      await mintAndApproveWrappedTokens(
+        wrappedToken,
+        lpRewards.address,
+        staker1,
+        wrappedTokenStakerBallance
+      )
+
+      await lpRewards.setRewardDistribution(rewardDistribution)
+      await lpRewards.notifyRewardAmount(keepRewards, {from: rewardDistribution})
+
+      const wrappedTokensToStake = web3.utils
+        .toBN(2000)
+        .mul(tokenDecimalMultiplier)
+      await lpRewards.stake(wrappedTokensToStake, {from: staker1})
+
+      const future = (await time.latest()).add(time.duration.days(7))
+      await timeIncreaseTo(future)
+
+      const actualEarnings = (await lpRewards.earned(staker1)).div(
```

Comment:
I'm not a fan of doing `.div(tokenDecimalMultiplier)` and `.mul(tokenDecimalMultiplier)` in so many places. Could we define values at the begining on the test including `tokenDecimalMultiplier` and don't use it aferwards?

Reply:
I know it doesn't look pretty, but I think there is a value when testing with `tokenDecimalMultiplier` which reflects the scenario when working with real tokens. When transferring tokens to a contract and then retrieving them, we need to divide by `10**18` which will also show a precision error that we're dealing with. This is a similar approach in other tests for rewards, ex. `TestECDSARewards.js`.
How about instead of multiplication we would use `web3.utils.toWei()` function 4e8d014?